compiler-rt/test/profile/Linux/binary-id-offset.c

```diff
@@ -5,28 +5,30 @@
 // (The DYN case would also apply to libraries, not explicitly tested here.)
 
 // DEFINE: %{cflags} =
-// DEFINE: %{check} = (                                                           \
-// DEFINE:     %clang_profgen -fuse-ld=lld -Wl,--build-id -o %t %s %{cflags}   && \
-// DEFINE:     env LLVM_PROFILE_FILE=%t.profraw %run %t                        && \
-// DEFINE:     llvm-readelf --notes %t                                         && \
-// DEFINE:     llvm-profdata show --binary-ids %t.profraw                         \
-// DEFINE:   ) | FileCheck %s
 
 // REDEFINE: %{cflags} = -no-pie
-// RUN: %{check}
+// RUN: %clang_profgen -fuse-ld=lld -Wl,--build-id -o %t %s %{cflags}
+// RUN: env LLVM_PROFILE_FILE=%t.profraw %run %t
+// RUN: llvm-readelf --notes %t && llvm-profdata show --binary-ids %t.profraw > FileCheck
```

Comment:
These are now just writing to a file called `FileCheck` instead of invoking the `FileCheck` command?

Reply:
Can this lit internal shell still do grouping? It would need something like `(llvm-readelf ... && llvm-profdata ...) | FileCheck %s` for the current `CHECK` lines to work. If not, you could split them into separate `--check-prefix` commands.

Reply:
> RUN: llvm-readelf --notes %t > %t2 && llvm-profdata show --binary-ids %t.pr\ofraw >> %t2 | FileCheck %t2

Try `... && FileCheck %s <%t2`? It needs the current file (`%s`) for the CHECK lines, and the test output on stdin.

Reply:
> Can this lit internal shell still do grouping?

It doesn't support subshells. The way we have been handling this is to have multiple commands write out to a file and then pass that to FileCheck using `--input-file`.